Sources/SwiftParser/Parameters.swift

```diff
@@ -92,10 +92,39 @@ extension Parser {
     let modifiers = parseParameterModifiers(isClosure: false)
     let misplacedSpecifiers = parseMisplacedSpecifiers()
 
-    let names = self.parseParameterNames()
-    let (unexpectedBeforeColon, colon) = self.expect(.colon)
-
-    let type = self.parseType(misplacedSpecifiers: misplacedSpecifiers)
+    var names = self.parseParameterNames()
+    var (unexpectedBeforeColon, colon) = self.expect(.colon)
+    
+    let type: RawTypeSyntax
+    
+    if colon.presence == .missing, let secondName = names.secondName, secondName.tokenText.isStartingWithUppercase {
+      // Synthesize the secondName parameter as a type node.
+      type = RawTypeSyntax(
+        RawSimpleTypeIdentifierSyntax(
+          name: RawTokenSyntax(
+            kind: .identifier,
+            text: secondName.tokenText,
+            presence: .missing,
+            arena: self.arena
+          ),
```

Comment:
You should be able to just use `secondName` here. There’s no need to construct a new `RawTokenSyntax`. That way, you also retain any trivia that’s attached to `secondName`. 

```suggestion
          name: secondName,
```

If you do that, I think you also don’t need to add `names.secondName` to `unexpectedBeforeColon`. And you might also not need to write any custom code in `ParseDiagnsoticGenerator` at all.

Reply:
Worked pretty well, thank you.